IBM J9 JDK/JRE is incorrectly flagged as not meeting minimum Java 8 update 251 requirement

**Describe the bug**
IBM J9 JDK/JRE implementing Java 8 update 251 or higher is incorrectly flagged as not meeting requirement for Java 8 update 251 or higher.

**Version of dependency-check used**
The problem occurs using version `9.0.4` of the maven plugin. The following output uses Java 8 `1.8.0_381` but the same version string issue remains with the current IBM JDK download linked below.

**Log file**
With context below, a log file (hopefully) shouldn't be necessary. Some selected output of `java -XshowSettings`, though, include:

```
...
Property settings:
    java.fullversion = 8.0.8.10 - pwa6480sr8fp10-20230703_02(SR8 FP10)
JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20230628_53798 (JIT enabled, AOT enabled)
OpenJ9   - a962f72
OMR      - 40dbd2d
IBM      - 696e9df
    java.runtime.name = Java(TM) SE Runtime Environment
    java.runtime.version = 8.0.8.10 - pwa6480sr8fp10-20230703_02(SR8 FP10)
    java.specification.maintenance.version = 5
    java.specification.name = Java Platform API Specification
    java.specification.vendor = Oracle Corporation
    java.specification.version = 1.8
    java.vendor = IBM Corporation
    java.version = 1.8.0_381
    java.vm.info = JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20230628_53798 (JIT enabled, AOT enabled)
OpenJ9   - a962f72
OMR      - 40dbd2d
IBM      - 696e9df
    java.vm.name = IBM J9 VM
    java.vm.specification.name = Java Virtual Machine Specification
    java.vm.specification.vendor = Oracle Corporation
    java.vm.specification.version = 1.8
    java.vm.vendor = IBM Corporation
    java.vm.version = 2.9
...
```

**To Reproduce**
Steps to reproduce the behavior:
1. Install and use a non-Semeru IBM J9 JDK or JRE for Java 8 with update >= 251
2. With a valid Maven POM/project attempt to execute any goal of `dependency-check-maven-plugin` in a usual manner (full coordinates, included in POM, etc.)
* Example: `mvn dependency-check:aggregate`
3. See error (some lines omitted/values redacted):

```
C:\[REDACTED]>mvn -version
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)                    
Maven home: [REDACTED]\apache-maven
Java version: 1.8.0_381, vendor: IBM Corporation, runtime: C:\[REDACTED]\java\8.0\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

C:\[REDACTED]>mvn dependency-check:aggregate
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
...
[INFO] Building [REDACTED]                          [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- dependency-check-maven:9.0.4:aggregate (default-cli) @ [REDACTED] ---
[ERROR] Non-supported Java Runtime: dependency-check requires at least Java 8 update 251 or higher.
```

**Expected behavior**
With the installed JVM implementing Java 8 `1.8.0_381`, the plugin runs and proceeds to generate a report in `target/` as normal.

**Additional context**
While common that `java.runtime.version` contains the desired value, IBM appears to differentiate the _runtime_ version from the Java version itself.

Oracle's naming convention referenced below in _Links_ might indicate that it would be more appropriate for [`org.owasp.dependencycheck.utils.Utils#getJavaUpdateVersion`](https://github.com/jeremylong/DependencyCheck/blob/f14e0f99f9211f02308ddedad63d9aa605b42d84/core/src/main/java/org/owasp/dependencycheck/utils/Utils.java#L66) to reference the `java.version` system property rather than `java.runtime.version`.

**Links**
[J2SE SDK/JRE Version String Naming Convention](https://www.oracle.com/java/technologies/javase/versioning-naming.html)
[IBM Java SDK Downloads 8.0](https://www.ibm.com/support/pages/java-sdk-downloads-version-80)

--- a/core/src/main/java/org/owasp/dependencycheck/utils/Utils.java
+++ b/core/src/main/java/org/owasp/dependencycheck/utils/Utils.java
@@ -63,7 +63,7 @@ public final class Utils {
      */
     public static int getJavaUpdateVersion() {
         //"1.8.0_144" "11.0.2+9" "17.0.8.1"
-        String runtimeVersion = System.getProperty("java.runtime.version");
+        String runtimeVersion = System.getProperty("java.version");
         return parseUpdate(runtimeVersion);
     }
 

--- a/core/src/test/java/org/owasp/dependencycheck/utils/UtilsTest.java
+++ b/core/src/test/java/org/owasp/dependencycheck/utils/UtilsTest.java
@@ -18,10 +18,10 @@
 package org.owasp.dependencycheck.utils;
 
 import org.junit.Test;
+
 import static org.junit.Assert.*;
 
 /**
- *
  * @author Jeremy Long
  */
 public class UtilsTest {
@@ -43,6 +43,11 @@ public class UtilsTest {
         assertEquals(expResult, result);
 
         runtimeVersion = "11.0.2+9";
+        expResult = 2;
+        result = Utils.parseUpdate(runtimeVersion);
+        assertEquals(expResult, result);
+
+        runtimeVersion = "11.0.2";
         expResult = 2;
         result = Utils.parseUpdate(runtimeVersion);
         assertEquals(expResult, result);

